firmware/Inc/PAL/GPIO.h

```diff
@@ -1 +1,23 @@
-#pragma once
+#ifndef GPIO_H
+#define GPIO_H
+
+#include "stm32l4xx_hal.h"
+
+namespace pal {
+    
+	class Gpio {
+	
+	private:
+		GPIO_TypeDef* port_;
+		uint16_t pin_;
+		
+	public:
+		Gpio(GPIO_TypeDef* port, uint16_t pin);
+		void Set(bool value) const;
```

Comment:
Also, according to that link:

> Accessors and mutators (get and set functions) may be named like variables

It doesn't say they must be or should be, and when I changed the function names to look like variables, it made the style just look bad and inconsistent. For example:

`Motor(pal::Gpio& step, pal::Gpio& dir, pal::Tim& tick);`
`int get_steps() const;`
`void reset_steps();`
`void SetDirection(Direction d);`
`void Step();`
`void SetAngularVelocity(float w, float radPerStep);`
`void SetMotion(bool on);`

It just feels inconsistent and i'd like to keep it at least in camel case